.github/workflows/build.yml

```diff
@@ -2,34 +2,89 @@ name: Build
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
-    branches: [ main ]
+    branches: [main]
 
 jobs:
-
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
+      - uses: actions/checkout@v4
+
+      - name: Run Gosec Security Scanner
+        uses: securego/gosec@master
+        with:
+          args: -exclude=G204 ./...
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.21"
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+
+      - name: Test
+        run: go test -v ./...
+
+      - name: Build
+        run: go build
+
+  demo:
```

Comment:
As I mention in the main PR comment, I'm happy to remove this job if you'd prefer it not cluster the `build.yml` workflow.

Reply:
I would like to keep it in separate file

Reply:
Sounds good; I've removed it!